src/Inventory.zig

```diff
@@ -2024,3 +2028,49 @@ pub fn loadFromZon(self: Inventory, zon: ZonElement) void {
 		}
 	}
 }
+
+fn toBase64(self: Inventory, allocator: NeverFailingAllocator) []const u8 {
+	var writer = BinaryWriter.init(main.stackAllocator);
+	defer writer.deinit();
+
+	self.toBytes(&writer);
+
+	const destination: []u8 = allocator.alloc(u8, std.base64.url_safe.Encoder.calcSize(writer.data.items.len));
+	return std.base64.url_safe.Encoder.encode(destination, writer.data.items);
+}
+
+fn toBytes(self: Inventory, writer: *BinaryWriter) void {
+	writer.writeVarInt(u32, @intCast(self._items.len));
+	for(self._items) |stack| {
+		stack.toBytes(writer);
+	}
+}
+
+fn fromBase64(self: Inventory, base64: []const u8) void {
+	const destination: []u8 = main.stackAllocator.alloc(u8, std.base64.url_safe.Decoder.calcSizeForSlice(base64) catch unreachable);
+	defer main.stackAllocator.free(destination);
+
+	std.base64.url_safe.Decoder.decode(destination, base64) catch unreachable;
+	var reader = BinaryReader.init(destination);
+	fromBytes(self, &reader);
+}
+
+fn fromBytes(self: Inventory, reader: *BinaryReader) void {
+	var count = reader.readVarInt(u32) catch 0;
+	for(self._items) |*stack| {
+		if(count == 0) {
+			stack.clear();
+			continue;
+		}
+		defer count -= 1;
```

Comment:
Since it doesn't make a difference, why not just do
```suggestion
		count -= 1;
```